connect/runtime/src/main/java/org/apache/kafka/connect/runtime/TransformationChain.java

```diff
@@ -32,10 +37,26 @@ public TransformationChain(List<Transformation<R>> transformations) {
     }
 
     public R apply(R record) {
+        return apply(record, NoopExecutor.INSTANCE, new ProcessingContext());
```

Comment:
wondering if there was a reason to leave this default method?

Reply:
leads to fewer changes in the codebase, easier to merge larger PRs.